### DISCUSSION TITLE
About Downloading Large Files

### DISCUSSION THREAD

**Discussion by hiro0025** [khoih-prog/FTPClient_Generic#3] [Q&A]
I use FTPClient _ Generic in Arduino MKRZER.
Downloading large files will not work properly.
Is there any way to save to SD card with FTPClient_Generic?
I use an Arduino to control various equipment. In order to monitor the solar system this time, I would like to periodically download the CSV files accumulated in the server.
The target file is 200 kb. Please advise.

**Comment by khoih-prog** [comment]
> Is there any way to save to SD card with FTPClient_Generic?

Sure there is many ways. But you have to master the SD library code, then write the application yourself to merge 2 libraries.

Check [**FTP_Server_SDFAT2**](https://github.com/khoih-prog/FTP_Server_Teensy41/tree/main/examples/FTP_Server_SDFAT2) example to understand how to deal with SD Cards.

There are too many public SD-related libraries now, for example

- [**RP2040_SD**](https://github.com/khoih-prog/RP2040_SD)
- [**SdFat**](https://github.com/greiman/SdFat)
- [**SD**](https://github.com/adafruit/SD)

It's better to post your issue, if you cant figure out how to do or to solve your issue, on **Arduino Forum**

Sorry, here is not a **general Forum** for every issue, only issues (**bug, enhancement, HOWTO USE**, etc) related to this library are suitable.

Good Luck,

**Reply by hiro0025** [reply]
I am sorry for the inconvenience caused by my unfamiliarity with this exchange.
I'll try it myself.
Thank you very much.
